test/extensions/transport_sockets/tls/cert_validator/default_validator_integration_test.cc

```diff
@@ -64,8 +64,8 @@ TEST_P(SslCertValidatorIntegrationTest, CertValidated) {
   codec->close();
 }
 
-// With verify-depth set, certificate validation fails
-TEST_P(SslCertValidatorIntegrationTest, CertValidationFailed) {
+// With verify-depth set, certificate validation succeeds since we allow partial chain
+TEST_P(SslCertValidatorIntegrationTest, CertValidatedWithVerifyDepth) {
```

Comment:
It seems `max_verify_depth` doesn't work well with `X509_V_FLAG_PARTIAL_CHAIN`.
Here the depth is set to 1, and intermediate_partial_ca_cert_chain.pem is set to trusted ca, which contains intermediate_ca_3  and intermediate_ca_2, the certificate is signed by intermediate_ca_2, see
https://github.com/envoyproxy/envoy/blob/bfc2cc21b19f9d2b4beb44d6f3cc2a9fa9ed4afd/test/integration/ssl_utility.cc#L50-L56

The intermediate_ca_2 depth is 2 in the chain so I thought it should fail if specifying depth to 1, but it succeeds.

This makes me confusing, do you have any ideas about this?

Reply:
Whoops, I misunderstood the `max_verify_depth`, it does not mean the depth of cert chain from the trust_ca file, it actually is the maximum number of intermediate certificate issuers. So this test's behavior is correct.